clang/lib/Sema/SemaType.cpp

```diff
@@ -7525,25 +7525,21 @@ static Attr *getCCTypeAttr(ASTContext &Ctx, ParsedAttr &Attr) {
 
 std::optional<FunctionEffectMode>
 Sema::ActOnEffectExpression(Expr *CondExpr, StringRef AttributeName) {
-  auto BadExpr = [&]() {
-    Diag(CondExpr->getExprLoc(), diag::err_attribute_argument_type)
-        << ("'" + AttributeName.str() + "'") << AANT_ArgumentIntegerConstant
-        << CondExpr->getSourceRange();
+  if (DiagnoseUnexpandedParameterPack(CondExpr))
     return std::nullopt;
-  };
-
-  if (CondExpr->isTypeDependent() || CondExpr->isValueDependent()) {
-    if (CondExpr->containsUnexpandedParameterPack())
-      return BadExpr();
+  if (CondExpr->isTypeDependent() || CondExpr->isValueDependent())
     return FunctionEffectMode::Dependent;
-  }
 
   std::optional<llvm::APSInt> ConditionValue =
       CondExpr->getIntegerConstantExpr(Context);
-  if (!ConditionValue)
-    return BadExpr();
-  return ConditionValue->getExtValue() ? FunctionEffectMode::True
-                                       : FunctionEffectMode::False;
+  if (!ConditionValue) {
+    Diag(CondExpr->getExprLoc(), diag::err_attribute_argument_type)
+        << ("'" + AttributeName.str() + "'") << AANT_ArgumentIntegerConstant
```

Comment:
I’d include the `''` in the diagnostic template instead of creating a string here.

Reply:
The difficulty here is:
- this is a pre-existing diagnostic; I didn't see a need for a new one just for quotes
- the other users  of the pre-existing diagnostic use `<<` with a `ParsedAttr`, which apparently quotes the attribute (?), but this function can be called from `TreeTransform` (for template instantiation), where there is only a `FunctionEffect` (whose name is passed in).

And then there are usages like:
```
    Diag(AttrLoc, diag::err_attribute_argument_type)
        << "vector_size" << AANT_ArgumentIntegerConstant
        << SizeExpr->getSourceRange();
```
, passing a string constant; it would appear that in this case the attribute isn't quoted, which seems inconsistent.

Reply:
> it would appear that in this case the attribute isn't quoted, which seems inconsistent.

Afaik types that are used to describe names of entities are automatically quoted. If you can’t access the name of the attribute or sth in that vein here, one option I think would be to get an `IdentifierInfo*` for the name and pass that in instead (`getASTContext().Idents.get("foobar")`).

Reply:
> passing a string constant

Actually, yes, that is inconsistent; it should be quoted because it is a language entity.

Reply:
So yeah, honestly, if it’s already inconsistent, then I’d just pass in the name w/o quotes here and leave fixing that everywhere to a subsequent pr.

Reply:
I did that, with a FIXME about the inconsistency of that diagnostic.